components/sbm-core/src/test/java/org/springframework/sbm/project/resource/TestProjectContext.java

```diff
@@ -575,18 +568,23 @@ private Integer getOrder(ProjectResourceWrapper l1) {
         @NotNull
         private ProjectContextInitializer createProjectContextInitializer() {
             AtomicReference<ProjectContextInitializer> projectContextInitializerRef = new AtomicReference<>();
-            if(beanFactory != null) {
+            if (beanFactory != null) {
                 ProjectContextInitializer bean = beanFactory.getBean(ProjectContextInitializer.class);
                 projectContextInitializerRef.set(bean);
                 executionContext = beanFactory.getBean(ExecutionContext.class);
             } else {
-                Map<Class<?>, Object> replacedBean = new HashMap<>();
-                if(sbmApplicationProperties != null) {
-                    replacedBean.put(SbmApplicationProperties.class, sbmApplicationProperties);
+                Map<Class<?>, Object> replacedBeans = new HashMap<>();
+                if (sbmApplicationProperties != null) {
+                    replacedBeans.put(SbmApplicationProperties.class, sbmApplicationProperties);
+                }
+
+                if (eventPublisher != null) {
+                    replacedBeans.put(ApplicationEventPublisher.class, eventPublisher);
                 }
 
-                if(eventPublisher != null) {
-                    replacedBean.put(ApplicationEventPublisher.class, eventPublisher);
+                // Test performance improvement on Windows
+                if (SystemUtils.IS_OS_WINDOWS) {
+                    replacedBeans.put(MavenPomCache.class, new InMemoryMavenPomCache());
```

Comment:
Let's not dynamically change behavior in a test-harness.
The performance change largely depends on the state of the `MavenPomCache` (filesystem when using RocksDB). 
If it has the dependencies it's faster, otherwise it might even be even slower.

Nevertheless, it would be great to find a reliable way to avoid the need for downloads during test runs, but the cached poms have additional metadata and thus can't be taken from local Maven repo. Whereas the RocksDB approach sometimes gave trouble with locks when used in tests.

Reply:
Wow, that's a lot. Did I get you right that the ÌnMemoryCache` dropped your build-time?